marketplacetests/tests/test_marketplace_incorrect_pin.py

```diff
@@ -0,0 +1,44 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from fxapom.fxapom import FxATestAccount
+
+from marketplacetests.marketplace_gaia_test import MarketplaceGaiaTestCase
+from marketplacetests.marketplace.app import Marketplace
+from marketplacetests.payment.app import Payment
+
+
+class TestMarketplaceIncorrectPin(MarketplaceGaiaTestCase):
+
+    def test_incorrect_pin(self):
+
+        APP_NAME = 'Test Zippy With Me'
+        good_pin = '1234'
+        bad_pin = '1111'
+        acct = FxATestAccount(base_url=self.base_url).create_account()
+
+        if self.apps.is_app_installed(APP_NAME):
+            self.apps.uninstall(APP_NAME)
+
+        marketplace = Marketplace(self.marionette, self.MARKETPLACE_DEV_NAME)
+        marketplace.launch()
+
+        marketplace.login(acct.email, acct.password)
+
+        marketplace.set_region('United States')
+
+        details_page = marketplace.navigate_to_app(APP_NAME)
+        details_page.tap_install_button()
+
+        payment = Payment(self.marionette)
+        payment.create_pin(good_pin)
+        payment.wait_for_buy_app_section_displayed()
+        self.assertIn(APP_NAME, payment.app_name)
+        payment.tap_cancel_button()
+
+        marketplace.wait_for_notification_message_displayed('Payment cancelled.')
```

Comment:
We need to complete the create PIN dialog in order to register the correct PIN, which brings us to a screen where we either have to `Buy` or `Cancel`. We then need to bring up the enter PIN dialog in order to test the invalid PIN. It seemed simpler to `Cancel` at this point rather than `Buy`. I haven't tested it, but I was also guessing that if we successfully Buy the app then we won't get an opportunity to enter our PIN again for purchase of this app, so we'd need yet another app to test with. Does that all make sense?